pkg/costmodel/costmodel.go

```diff
@@ -1075,27 +1075,38 @@ func (cm *CostModel) GetNodeCost(cp costAnalyzerCloud.Provider) (map[string]*cos
 		// not all providers are guaranteed to use this, so don't overwrite a Provider assignment if we can't find something under that capacity exists
 		gpuc := 0.0
 		q, ok := n.Status.Capacity["nvidia.com/gpu"]
+		gpuCount := q.Value()
 		_, hasReplicas := n.Labels["nvidia.com/gpu.replicas"]
 
-		if ok && !hasReplicas {
-			gpuCount := q.Value()
-			if gpuCount != 0 {
-				newCnode.GPU = fmt.Sprintf("%d", gpuCount)
-				newCnode.VGPU = newCnode.GPU
-				gpuc = float64(gpuCount)
-			}
-		} else if hasReplicas { // See https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html
-			if q.Value() == 0 {
-				q = n.Status.Capacity["nvidia.com/gpu.shared"]
-			}
+		if ok && gpuCount != 0 && !hasReplicas {
+			newCnode.GPU = fmt.Sprintf("%d", gpuCount)
+			newCnode.VGPU = newCnode.GPU
+			gpuc = float64(gpuCount)
+		} else if hasReplicas {
+			// Special case. The "nvidia.com/gpu.replicas" label is usually
+			// applied by NVIDIA's GPU Operator or GPU Feature Discovery pod.
+			// Ref: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html#verifying-the-gpu-time-slicing-configuration
+			// Ref: https://github.com/NVIDIA/k8s-device-plugin/blob/d899752a424818428f744a946d32b132ea2c0cf1/internal/lm/resource_test.go#L44-L45
+			// Ref: https://github.com/NVIDIA/k8s-device-plugin/blob/d899752a424818428f744a946d32b132ea2c0cf1/internal/lm/resource_test.go#L103-L118
+
 			g, ok := n.Labels["nvidia.com/gpu.count"]
 			if ok {
 				newCnode.GPU = g
+				gpuc, err = strconv.ParseFloat(g, 64)
+				if err != nil {
+					log.Warnf("Could not parse label \"nvidia.com/gpu.count\": \"%s\". Setting to 0.", g)
+					gpuc = 0.0
+				}
 			} else {
-				newCnode.GPU = fmt.Sprintf("%d", 0)
+				log.Warnf("Could not find label \"nvidia.com/gpu.count\". Setting to 0.")
```

Comment:
We now have a warning here, but seem to have removed the line setting to `0`. Is that intentional?

Reply:
Good callout! This was fixed in my full refactor recently. Marking as resolved!